src/par_iter/mod.rs

```diff
@@ -609,18 +609,19 @@ pub trait ParallelIterator: Sized {
     fn drive_unindexed<C>(self, consumer: C) -> C::Result
         where C: UnindexedConsumer<Self::Item>;
 
+    /// Internal method used to fake specialization for indexed collect.
+    #[doc(hidden)]
+    fn opt_len(&mut self) -> Option<usize> { None }
```

Comment:
Let's say a bit more here. It's not clear to me that this method needs to be hidden and internal, but it can't hurt to keep it that way to start:

"Returns the number of items produced by this iterator, if known statically. This can be used by consumers to trigger special fast paths. Therefore, if `Some(_)` is returned, this iterator must only use the (indexed) `Consumer` methods when driving a consumer, such as `split_at()`. Calling `UnindexedConsumer::split_off()` or other `UnindexedConsumer` methods -- or returning an inaccurate value -- may result in panics."